src/rtde/control_package_setup_outputs.cpp

```diff
@@ -34,16 +34,30 @@ namespace rtde_interface
 {
 bool ControlPackageSetupOutputs::parseWith(comm::BinParser& bp)
 {
-  bp.parse(output_recipe_id_);
-  bp.parseRemainder(variable_types_);
+  if (protocol_version_ == 2)
+  {
+    bp.parse(output_recipe_id_);
+    bp.parseRemainder(variable_types_);
+  }
+  else if (protocol_version_ == 1)
+  {
+    bp.parseRemainder(variable_types_);
+  }
```

Comment:
Maybe adding a plain `else` tree here if UR ever decides to to a protocol version 3 would be good. This way it will be obvious why, when things go south.

Reply:
That is a good idea, I have added a plain else statement.